### PR TITLE
PLATFORM-9039: webhooks reliability

### PR DESCRIPTION
**Pull request by korzepadawid**
## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

File: driver/registry_default.go [hunk 1]
```diff
@@ -816,11 +816,16 @@ func (m *RegistryDefault) PrometheusManager() *prometheus.MetricsManager {
...
+	m.rwl.RLock()
 	if cl, ok := m.rc[name]; ok {
+		m.rwl.RUnlock()
 		rc = cl
 	} else {
+		m.rwl.RUnlock()
 		rc = m.HTTPClient(ctx, opts...)
+		m.rwl.Lock()
 		m.rc[name] = rc
+		m.rwl.Unlock()
```
Comment (korzepadawid) [inline driver/registry_default.go:828]: there was a data race before

File: selfservice/hook/web_hook.go [hunk 2]
```diff
@@ -540,17 +538,28 @@ func (e *WebHook) execute(ctx context.Context, data *templateContext) error {
 	return nil
 }
 
-func parseWebhookResponse(resp *http.Response, id *identity.Identity) (err error) {
+func (e *WebHook) parseWebhookResponse(resp *http.Response, id *identity.Identity) (err error) {
```
Comment (korzepadawid) [inline selfservice/hook/web_hook.go:541]: let's try to push it to the upstream

Comment (korzepadawid) [reply selfservice/hook/web_hook.go:541]: otherwise we don't have an access to the logger

Comment (korzepadawid) [reply selfservice/hook/web_hook.go:541]: logger is only available through `e.deps`

File: selfservice/hook/web_hook_integration_test.go [hunk 3]
```diff
@@ -934,8 +936,8 @@ func TestWebHooks(t *testing.T) {
 		// error.
 
 		var wg sync.WaitGroup
-		wg.Add(3) // HTTP client does 3 attempts
```
Comment (korzepadawid) [inline selfservice/hook/web_hook_integration_test.go:937]: that's no longer true assuming this https://github.com/Wikia/kratos/pull/97/files#r1479563845

File: identity/handler.go [hunk 4]
```diff
@@ -736,72 +734,6 @@ func (h *Handler) validate(w http.ResponseWriter, r *http.Request, _ httprouter.
...
-//	Schemes: http, https
-//
-//	Security:
-//	  oryAccessToken:
-//
-//	Responses:
-//	  204: emptyResponse
-//	  404: jsonError
-//	  500: jsonError
-func (h *Handler) deleteMultifactorCredential(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
```
Comment (korzepadawid) [inline identity/handler.go:777]: context: https://github.com/Wikia/kratos/pull/97/files#r1478195394

File: driver/registry_default.go (hunk 1 above)
Comment (adpaste) [reply driver/registry_default.go:828]: nit: Could we use https://pkg.go.dev/sync#Map instead of manual synchronization?

**Pull request merged**